src/tests/Interop/common/ComHelpers.h

```diff
@@ -21,7 +22,12 @@ namespace Internal
         /* [iid_is][out] */ _COM_Outptr_ void __RPC_FAR *__RPC_FAR *ppvObject,
         /* [in] */ I obj)
     {
+#ifdef _WIN32        
         if (riid == __uuidof(I))
+#else
+        //__uuidof(I) doesn't work on linux, so needed types are checked
+        if (riid == __uuidof(IInspectable) || riid == __uuidof(IWeakReference) || riid == __uuidof(IWeakReferenceSource))
```

Comment:
I don't see how this is correct. This is a templated type and now it is a specialized type which makes it of dubious value. What is the problem here?

Reply:
Yes, here the problem is that on linux __uuidof(I) doesn't work. In the last meeting we discussed about the fact that only WeakReference prj uses this method so I though that we could check only for types we know are used with it. For sure is not a good solution, how could we proceed instead?

Reply:
Instead of manipulating the template implementation here I would provide one in the actual `QueryInterface` implementation. 

```cpp
{
#ifdef _WIN32
    return DoQueryInterface<...>();
#else
    // Provide custom implementation.
#endif // !_WIN32
}
```

Reply:
Ok , I've modified QueryInterface inside WeakReferenceNative.cpp and enabled Internal::__QueryInterfaceImpl related code only for _WIN32 in ComHelpers.h